dialog/file.go

```diff
@@ -54,7 +55,9 @@ type fileDialog struct {
 	showHidden       bool
 
 	view viewLayout
-	data []fyne.URI
+
+	data     []fyne.URI
+	dataLock *sync.RWMutex
```

Comment:
You can remove the pointer and just embed `sync.RWMutex` into the struct directly. It avoids having to manually create the mutex in all struct definitions below.

Reply:
Sounds good. I have updated the branch and pushed.